classes/local/upload_helper.php

```diff
@@ -76,10 +76,11 @@ public static function get_upload_jobs($courseid) {
 
         $allnamefields = get_all_user_name_fields(true, 'u');
 
-        $sql = "SELECT uj.*, f.filename, f.filesize, $allnamefields
+        $sql = "SELECT uj.*, f.filename, f.filesize, $allnamefields, md.metadata
                 FROM {block_opencast_uploadjob} uj
-                JOIN {files} f ON uj.fileid = f.id AND f.component = :component
+                JOIN {files} f ON uj.presenter_fileid = f.id OR uj.presentation_fileid = f.id AND f.component = :component
```

Comment:
Through this or you return multiple rows with the same uploadjob id. You should better JOIN to files twice. Once for the presenter and once for the presentation.

Reply:
In future extensions (or directly if you like to) we could then improve the table for existing upload_jobs, by showing the information for both files.